core/controllers/story_viewer.py

```diff
@@ -85,6 +128,26 @@ class StoryProgressHandler(base.BaseHandler):
     """
 
     GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS = {
+        'classroom_url_fragment': {
+            'schema': SCHEMA_FOR_CLASSROOM_URL_FRAGMENT
+        },
+        'topic_url_fragment': {
+            'schema': SCHEMA_FOR_TOPIC_URL_FRAGMENT
+        },
+        'story_url_fragment': {
+            'schema': SCHEMA_FOR_STORY_URL_FRAGMENT
+        },
+        'node_id': {
+            'schema': {
+                'type': 'basestring'
+            }
+        }
```

Comment:
Can more rules be introduce here?

Reply:
We can use this as a reference.
https://github.com/mahendra1290/oppia/blob/293831f30952d28774d1bd3b6c35da7b3eef84f7/core/domain/story_domain.py#L254
![Screenshot from 2021-09-24 22-53-45](https://user-images.githubusercontent.com/42772952/134715983-cd4a6fe7-e688-4911-9f60-ced895d096b7.png)

Reply:
agreed

Reply:
when using regex for validation some tests fail. As we are passing 'invalid_node' as node_id which is not a valid node id so we never get a 404 error and our tests fail.

![Screenshot from 2021-09-25 12-08-31](https://user-images.githubusercontent.com/42772952/134761584-ea967586-2d43-4ae8-a97f-e445e2c0673e.png)

![Screenshot from 2021-09-25 12-09-56](https://user-images.githubusercontent.com/42772952/134761624-b4d85b3a-f5cc-4794-99e2-b99d2f99ef8c.png)

Reply:
Yes you can modify there to raise 400 as we are expecting it to fail by Schema Validation System.

Reply:
done